sky/clouds/gcp.py

```diff
@@ -1136,6 +1342,112 @@ def _get_disk_specs(
             specs['disk_iops'] = 20000
         return specs
 
+    @classmethod
+    def _validate_instance_volumes(
+        cls,
+        instance_type: Optional[str],
+        volumes: Optional[List[Dict[str, Any]]],
+    ) -> None:
+        if not volumes:
+            return
+        if instance_type is None:
+            logger.warning(
+                'Instance type is not specified, skipping instance volume validation'
+            )
+            return
+        instance_volume_count = 0
+        for volume in volumes:
+            if volume['storage_type'] == resources_utils.StorageType.INSTANCE:
+                instance_volume_count += 1
+        if instance_type in constants.SSD_AUTO_ATTACH_MACHINE_TYPES and instance_volume_count > constants.SSD_AUTO_ATTACH_MACHINE_TYPES[
+                instance_type]:
+            raise exceptions.ResourcesUnavailableError(
+                f'The instance type {instance_type} supports {constants.SSD_AUTO_ATTACH_MACHINE_TYPES[instance_type]} instance storage, but {instance_volume_count} are specified'
+            )
+        # TODO(hailong):
+        # check the instance storage count for the other instance types,
+        # refer to https://cloud.google.com/compute/docs/disks/local-ssd#lssd_disk_options
+
+    @classmethod
+    def _get_volumes_specs(
```

Comment:
can we move some of the volume related code into `provision/gcp/volume_utils` to keep the cloud class cleaner?

Reply:
OK

Reply:
This function needs to access the other class methods, the others are moved.